sphinx_gallery/recommender.py

```diff
@@ -0,0 +1,269 @@
+# -*- coding: utf-8 -*-
+"""Recommendation system generator.
+
+Generate recommendations based on TF-IDF representation and a KNN model.
+"""
+# Author: Arturo Amor
+# License: 3-clause BSD
+
+
+import numbers
+import numpy as np
```

Comment:
... in other words, you need to nest this import where you need `np` rather than making it a top-level import

Reply:
... and whatever tests require `numpy` / recommender (e.g., `tinybuild` tests) you'll also want to add an `importorskip` too